hack/make/.binary

```diff
@@ -74,19 +74,11 @@ source "${MAKEDIR}/.go-autogen"
 		fi
 	fi
 
-	# This is a workaround to have buildinfo with deps embedded in the binary. We
-	# need to create a go.mod file before building with -modfile=vendor.mod,
-	# otherwise it fails with: "-modfile cannot be used to set the module root directory."
-	if [ ! -f "go.mod" ]; then
-		printf '%s\n\n%s' 'module github.com/docker/docker' 'go 1.19' > "go.mod"
-		trap 'rm -f go.mod' EXIT
-	fi
-
 	echo "Building $([ "$DOCKER_STATIC" = "1" ] && echo "static" || echo "dynamic") $DEST/$BINARY_FULLNAME ($PLATFORM_NAME)..."
 	if [ -n "$DOCKER_DEBUG" ]; then
 		set -x
 	fi
-	GO111MODULE=on go build -mod=vendor -modfile=vendor.mod -o "$DEST/$BINARY_FULLNAME" "${BUILDFLAGS[@]}" -ldflags "$LDFLAGS $LDFLAGS_STATIC $DOCKER_LDFLAGS" ${GO_PACKAGE}
+	./hack/with-go-mod.sh go build -mod=vendor -modfile=vendor.mod -o "$DEST/$BINARY_FULLNAME" "${BUILDFLAGS[@]}" -ldflags "$LDFLAGS $LDFLAGS_STATIC $DOCKER_LDFLAGS" "$GO_PACKAGE"
```

Comment:
Related to https://github.com/moby/moby/pull/45304#discussion_r1190880643 I presume? Output will still be clunky using this script: https://github.com/moby/moby/actions/runs/5520712208/jobs/10067746529#step:5:1426

```
#56 [build 6/6] RUN --mount=type=bind,target=.,rw     --mount=type=tmpfs,target=cli/winresources/dockerd     --mount=type=tmpfs,target=cli/winresources/docker-proxy     --mount=type=cache,target=/root/.cache/go-build,id=moby-build-linux/amd64 <<EOT (set -e...)
#56 1.120 
#56 1.394 Removing /tmp/bundles/
#56 1.401 
#56 1.411 ---> Making bundle: binary (in /tmp/bundles/binary)
#56 1.513 Building static /tmp/bundles/binary-daemon/dockerd (linux/amd64)...
#56 1.520 + tee /go/src/github.com/docker/docker/go.mod
#56 1.524 module github.com/docker/docker
#56 1.524 
#56 1.524 go 1.19
#56 1.525 + trap 'rm -f "${ROOTDIR}/go.mod"' EXIT
#56 1.526 + GO111MODULE=on
#56 1.528 + go build -mod=vendor -modfile=vendor.mod -o /tmp/bundles/binary-daemon/dockerd -tags 'netgo osusergo static_build  journald' -ldflags '-w -X "github.com/docker/docker/dockerversion.Version=pr-45929" -X "github.com/docker/docker/dockerversion.GitCommit=HEAD" -X "github.com/docker/docker/dockerversion.BuildTime=2023-07-11T13:44:08.000000000+00:00" -X "github.com/docker/docker/dockerversion.PlatformName=" -X "github.com/docker/docker/dockerversion.ProductName=" -X "github.com/docker/docker/dockerversion.DefaultProductLicense="  -extldflags -static ' github.com/docker/docker/cmd/dockerd
#56 ...
```

Can we get just `printf` here?: https://github.com/moby/moby/blob/43007108e65a01be5044262b3f3399c95bc9de46/hack/with-go-mod.sh#L23-L27

Like https://github.com/moby/moby/pull/44917/files#diff-fa876a8e859e8583907d556eeed2e4b8cc98d7847b2124521aaef1e9984f50cfR21

Reply:
I still like `set -x` and the `tee` as it makes it easy to see in the logs exactly what is happening, but it's not a hill I'll die on if you think it's that objectionable.

Reply:
I don't really mind :+1: